spec/features/deserialization_spec.rb

```diff
@@ -0,0 +1,59 @@
+require 'rails_helper'
+
+RSpec.feature "deserialization" do
+  include ActiveJobHelper
+  include RackTestFeaturesHelper
+
+  before do
+    Timecop.travel Time.new(2015, 9, 1, 15, 47).utc
+    Capybara.current_driver = :rack_test
+    VisitMetricsEntry.delete_all
+  end
+
+  after do
+    Timecop.return
+  end
+
+  let(:data) {
+    YAML.load(
+      File.read(
+        File.expand_path('../../fixtures/deserialization_data.yml', __FILE__)
+      )
+    )
+  }
+
+  context 'before removing deferred namespace' do
+    let(:data) {
+      super().fetch('before_removing_deferred_namespace')
+    }
+
+    before do
+      data.fetch('visit_metrics_entries').each do |hash|
+        VisitMetricsEntry.create! hash
+      end
+      allow_any_instance_of(ApplicationController).
+        to receive(:reject_untrusted_ips!)
+    end
+
+    scenario 'booking receipt' do
+      link = data.fetch('booking_receipt_path')
+      visit link
+      expect(status).to eq(200)
```

Comment:
Might be nice if these used [the `have_http_status` matcher](https://relishapp.com/rspec/rspec-rails/docs/matchers/have-http-status-matcher).